dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -331,6 +336,16 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
       }
     }
 
+    void clear() {
+      encoded_length = 0;
+      bpp = 8;
+      selected_decoder = nullptr;
+      is_progressive = false;
+      req_nchannels = -1;
+      method = DecodeMethod::Host;
+      subsampling = NVJPEG_CSS_UNKNOWN;
+    }
```

Comment:
*It's beyond the scope of this PR*, but this method (and the fact that you can't just assign `{}` to reset it) is a manifestation of an underlying problem: this object serves two purposes. It is both SampleData and SampleDecoderContext or whatever the name should be. I don't think they should be aggregated this way - perhaps a valid solution would be to rename it to SampleContext and create a subaggregate (SampleData) which would contain the fields that are reset here (+ filename - which should probably (?) be reset here, too).

Reply:
I agree with this. I'll add a TODO here for now. It deserves a separate PR.